extendr-api/src/conversions/try_into_int.rs

```diff
@@ -27,25 +28,27 @@ macro_rules! impl_into_integerish {
     ($float_type:ty, $int_type:ty) => {
         impl FloatToInt<$int_type> for $float_type {
             fn try_into_int(&self) -> Result<$int_type, ConversionError> {
-                if *self == <$float_type>::default() {
-                    return Ok(<$int_type>::default());
+                match self.classify() {
+                    FpCategory::Nan | FpCategory::Subnormal => Err(ConversionError::NotIntegerish),
+                    FpCategory::Zero => Ok(<$int_type>::default()),
+                    FpCategory::Infinite if self.is_sign_positive() => {
+                        Err(ConversionError::Overflow)
+                    }
+                    FpCategory::Infinite => Err(ConversionError::Underflow),
+                    FpCategory::Normal => {
+                        let truncated_value = self.trunc();
+                        if truncated_value < <$int_type>::MIN as $float_type {
+                            return Err(ConversionError::Underflow);
+                        }
+                        if truncated_value > <$int_type>::MAX as $float_type {
+                            return Err(ConversionError::Overflow);
+                        }
+                        if !truncated_value.eq(self) {
+                            return Err(ConversionError::NotIntegerish);
```

Comment:
I also want to ask this.. Why don't we `.clamp` it, and say that the conversion is following the Saturating mode, of https://doc.rust-lang.org/std/num/struct.Saturating.html ?

I'm not saying anything against what you've done. I just somehow, sometimes, feel like we have too many errors. Like just give the user _something_ even if the request was absurd.

Reply:
Because it will revert to precious behavior -- silently changing values https://github.com/extendr/extendr/issues/720#issue-2221351632

Reply:
> Because it will revert to precious behavior -- silently changing values [#720 (comment)](https://github.com/extendr/extendr/issues/720#issue-2221351632)

But isn't that _fine_?

Reply:
I would say absolutely not, one of the reasons people criticize `as` operators in Rust because of this sh1t. If you pass stupidly large value that does not fit in your target type, this is an error, not `i32::MAX`